src/vm/tests/costs.rs

```diff
@@ -427,3 +436,383 @@ fn test_cost_contract_short_circuits() {
+            "confirmed-proposal-count",
+            Value::UInt(bad_proposals as u128 + good_proposals),
+        )
+        .unwrap();
+
+        for (ix, (intercepted_ct, intercepted_f, cost_ct, cost_f)) in
+            good_cases.into_iter().enumerate()
+        {
+            let value = format!(
+                "{{ function-contract: '{},
+                    function-name: \"{}\",
+                    cost-function-contract: '{},
+                    cost-function-name: \"{}\",
+                    confirmed-height: u1 }}",
+                intercepted_ct, intercepted_f, cost_ct, cost_f
+            );
+            db.set_entry(
+                &STACKS_BOOT_COST_VOTE_CONTRACT,
+                "confirmed-proposals",
+                execute(&format!("{{ confirmed-id: u{} }}", ix + bad_proposals)),
+                execute(&value),
+            )
+            .unwrap();
+        }
+        db.commit();
+    }
+
+    marf_kv.test_commit();
+    marf_kv.begin(&StacksBlockId([4 as u8; 32]), &StacksBlockId([5 as u8; 32]));
+
+    {
+        let mut owned_env = OwnedEnvironment::new_max_limit(
+            marf_kv.as_clarity_db(&NULL_HEADER_DB, &NULL_BURN_STATE_DB),
+        );
+
+        execute_transaction(
+            &mut owned_env,
+            p2.clone(),
+            &caller,
+            "execute",
+            &symbols_from_values(vec![Value::UInt(5)]),
+        )
+        .unwrap();
+
+        let (_db, tracker) = owned_env.destruct().unwrap();
+
+        let circuits = tracker.contract_call_circuits();
+        assert_eq!(circuits.len(), 2);
+
+        let circuit1 = circuits.get(&(intercepted.clone(), "intercepted-function".into()));
+        let circuit2 = circuits.get(&(intercepted.clone(), "intercepted-function2".into()));
+
+        assert!(circuit1.is_some());
+        assert!(circuit2.is_some());
+
+        assert_eq!(circuit1.unwrap().contract_id, cost_definer);
+        assert_eq!(circuit1.unwrap().function_name, "cost-definition");
+
+        assert_eq!(circuit2.unwrap().contract_id, cost_definer);
+        assert_eq!(circuit2.unwrap().function_name, "cost-definition-multi-arg");
+
+        for (target, referenced_function) in tracker.cost_function_references().into_iter() {
+            if target == &ClarityCostFunction::Le {
+                assert_eq!(&referenced_function.contract_id, &cost_definer);
+                assert_eq!(&referenced_function.function_name, "cost-definition-le");
+            } else {
+                assert_eq!(
+                    &referenced_function.contract_id, &*STACKS_BOOT_COST_CONTRACT,
+                    "Cost function should still point to the boot costs"
+                );
+                assert_eq!(
+                    &referenced_function.function_name,
+                    target.get_name_str(),
+                    "Cost function should still point to the boot costs"
+                );
+            }
+        }
+    };
+
+    marf_kv.test_commit();
```

Comment:
Does it make any sense to check the `ExecutionCost` after evaluating each contract-call with the newly-applied cost function, in order to verify that the cost function is getting invoked?

Reply:
It's tested elsewhere as well, but I added it here as well in 926383028